app/transport/rest/user.go

```diff
@@ -96,13 +122,51 @@ func (uh UserHandler) GetUserByID() http.Handler {
 			return
 		}
 
-		user := User{
-			ID:        entUser.ID,
-			FirstName: entUser.FirstName,
-			LastName:  entUser.LastName,
-			Email:     entUser.Email,
-			CreatedAt: entUser.CreatedAt,
+		userResp := User{
+			ID:        user.ID,
+			FirstName: user.FirstName,
+			LastName:  user.LastName,
+			Email:     user.Email,
+			CreatedAt: user.CreatedAt,
 		}
-		WriteJSONResponse(w, http.StatusOK, user, uh.logger)
+		WriteJSONResponse(w, http.StatusOK, userResp, uh.logger)
+	})
+}
+
+// QueryParams represent request query params.
+type QueryParams struct {
+	Offset string `schema:"offset"`
+	Limit  string `schema:"limit"`
+	Sort   string `schema:"sort"`
+}
+
+// GetUsers retrieves all entities.User by given parameters.
+func (uh UserHandler) GetUsers() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		var params QueryParams
+
+		if err := schema.NewDecoder().Decode(&params, req.URL.Query()); err != nil {
+			uh.logger.Errorf("Failed parsing query %+v: %+v", req.URL.Query(), err)
+			WriteJSONResponse(w, http.StatusInternalServerError, Response{Message: MsgInternalSeverErr, Details: "could not parse query"}, uh.logger)
+
+			return
+		}
+
+		users, err := uh.usecase.GetAll(params.Offset, params.Limit, params.Sort)
+		if err != nil {
+			uh.logger.Errorf("Failed fetching users from repository: %+v", err)
+
+			if errors.Is(err, globals.ErrBadQuery) {
+				WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: globals.ErrBadQuery.Error()}, uh.logger)
```

Comment:
What do you mean by bad query?
You must do validation before making any request in the very beginning of the handler.

Reply:
We were also recommended to remove query validation from the routing. I completely removed it to make things simpler

Reply:
Validation and `QueryParam`  were moved to `entities` package.